doc/contribute/index.rst

```diff
@@ -130,6 +130,20 @@ to add the sign-off you can also amend a previous commit with the sign-off by
 running ``git commit --amend -s``. If you've pushed your changes to GitHub
 already you'll need to force push your branch after this with ``git push -f``.
 
+Notes
+=====
+
+Developers are permitted to cherry-pick patches that are included in
+pull requests submitted by other contributors. It is expected that
+
+* the content of the patches should not be modified
+* the cherry-picked commits shall preserve the original sign-off messages and
+  the author identity.
```

Comment:
A real-world application of this is #25971.

The first commit contains some fragments that were unchanged from the [downstream commit](https://github.com/nrfconnect/sdk-zephyr/pull/305/commits/5500a99cc02a1675663a9126bbd71e5c248c750c), but the bulk of the material is new.    I kept the SOB line from @sigvartmh to acknowledge that fragments were taken unmodified (so asserting the origin for what I kept is valid), but submitted it under my authorship---in part because I completely rewrote the commit message and felt that implying he had primary responsibility was inappropriate.

I'm willing to engage in discussion about whether this was the best way to proceed.  AFAIK it was found acceptable by the original author.

Reply:
Small comment about this. When we cherry-picks any commit and keep it untouched we know who made the contribution. When we change anything, we don't know who made what. I think this could hurt people and create unnecessary discussions and worst, people stop send contributions. If a cherry-picks is important we need keep as original (even with bugs). If we need change a work, it would be cherry-picked without any modifications and then create another commit with the changes and say why. That makes things clear instead blurry.

Reply:
@nandojve thanks for your comment: I think your concern is reflected in the document. To summarize
- cherry-picking without content change is allowed and should leave commit  author/sign-off as is
- pick & modify leaves 2 options
   - reach out to original author to ACK
   - send the patch with own author/sign-off and leave a note about source of inspiration (optional)

> it would be cherry-picked without any modifications and then create another commit with the changes and say why.

@nandojve this was actually discussed and , although it is a valid option, it is not always ideal, since this creates extra noise in PR reviews but, most importantly, it affects bisectability (in case it contains obvious bugs)

Reply:
> pick & modify leaves 2 options
> * reach out to original author to ACK
> * send the patch with own author/sign-off and leave a note about source of inspiration (optional)

Small changes like adjusts to fit: add below is enough in my opinion because it's a clean reason and we need keep tracking of what happened somehow.
```
[Project Maintainer: Renamed test methods according to naming convention.]
Signed-off-by: Project Maintainer <project.maintainer@example.com>
```

The huge changes that imply own the code we **must** communicate original Author not "should" as mentioned. It seems section **Recommended policies** is not useful because say that anyone can get anything and do whatever they want without say anything to anyone. One thing is the end object zephyr project it self, the other thing is the sequence (commits) that makes the zephyr project. There is no Apache Licence on every commit. So, when someone takes code from others and do a lot of changes it must communicate, it must explain the why, it should invite to review, it should ask Author ack. We shouldn't give super powers to anyone, we should propose a contract that is fair to every one.

Take this for example:
> Any contributions made as part of submitted pull requests are considered free

True, as is. Above don't give rights to change the inside content of a commit.

I hope above not sounds like a complain. My point is, we need think there are people sending code, their deserves respect and they have feelings. If Zephyr don't want to be a dictator it must have policy to avoid that.

Reply:
>The huge changes that imply own the code we must communicate original Author not "should" as mentioned. It seems section Recommended policies is not useful because say that anyone can get anything and do whatever they want without say anything to anyone.

> So, when someone takes code from others and do a lot of changes it must communicate, it must explain the why, it should invite to review, it should ask Author ack. We shouldn't give super powers to anyone, we should propose a contract that is fair to every one.

@nandojve thanks for stressing this.
So, I think all this is present in the text; I believe you propose to make this more of a requirement, not a recommendation. So I am thinking of
- changing "Recommended policies" to "Accepted policies"
- changing **should** to **shall** in:
    - explain why
    - invite original author to the review
    - ask for ACK from them

Would these changes satisfy your concerns here?

Reply:
@ioannisg yes! I believe most important is explain and communicate. With that, no one have space to complain. The invide is the best way to communicate because shows that Zephyr cares about what is going on. It is up to Author ACK and participate on the review process. However, we need care to give time to Author manifest. This should follow normal review process and it may be important reinforce that.

Reply:
that's all fine, @nandojve , I did the proposed updates, please, take a look.

Reply:
It seems like the easiest way to deal with cherry-picking is to cherry-pick the changes as standalone commits in your PR then add your changes on top of them as separate commits where your commit message will identify why you made changes (and maybe why you cherry-picked a different PR instead of helping to push that PR to merge).